src/main/resources/META-INF/resources/primefaces/menu/menu.menubar.js

```diff
@@ -43,11 +43,18 @@ PrimeFaces.widget.Menubar = PrimeFaces.widget.TieredMenu.extend({
             clearTimeout(this.timeoutId);
         }
 
-        this.timeoutId = setTimeout(function () {
-           submenu.css('z-index', PrimeFaces.nextZindex())
-                  .show()
-                  .position(pos)
-        }, this.cfg.delay);
+        //avoid using timeout if delay is 0
+        if(this.cfg.delay != null && this.cfg.delay > 0) {
```

Comment:
You can change this to...
```javascript
if(this.cfg.delay && this.cfg.delay > 0) {
```

This will be false if delay ==0 or delay == null or delay == undefined.

Reply:
Thanks for the feedback. I've updated the change accordingly.